Complete MongoDB mapper implementation

a-type opened the pull request:
Hope this isn't too invasive.

I think Mongo needs to be set up in Travis as well.

jagoda committed:
Create instance in MongoDB.

jagoda committed:
Look up instances in MongoDB.

jagoda committed:
Update existing instances in MongoDB.

a-type committed:
Complete MongoDB mapper implementation

a-type committed:
Automatic revision field on Model instances

jagoda commented on lib/mappers/mongo.js:
```diff
@@ -95,7 +95,13 @@ function MongoMapper () {
 				query[index] = instance[index];
 
-				return collection.updateAsync(query, instance);
+				return collection.findOneAsync(query)
```
Is this really necessary? I would expect MongoDB to somehowe notify us that no documents were modified (however I recall we have had some issues with this). Maybe this could help: http://docs.mongodb.org/manual/reference/method/db.collection.update/#writeresult

jagoda commented on lib/mappers/mongo.js:
```diff
@@ -95,7 +95,13 @@ function MongoMapper () {
+					if (!findResult) {
+						throw new Error("The document you tried to update does not exist.");
+					}
+					return collection.updateAsync(query, instance);
```
This really needs revision IDs to work correctly.

jagoda commented:
This looks really good! Thanks for following the spirit of what was already there. A few additional pieces of learning that we should probably incorporate:
1. Connection pooling. Seems my initial approach with disposers here is somewhat of an anti-pattern (see http://blog.mongolab.com/2013/11/deep-dive-into-connection-pooling/).
2. We should probably close the cursor before returning from the `find` method.
3. We should handle concurrent updates is some reasonable way.

jagoda commented:
Also, we should support a custom DB URL rather than always using the default (otherwise you may find this difficult to use in production ;-) ).

a-type committed:
Concurrency checking using revision numbers in mappers

a-type committed:
Destroy operations for mongo mapper

a-type committed:
Close cursor after findAll operation

a-type committed:
Persistent DB connection

a-type committed:
Custom DB URL

a-type committed:
Add MongoDB to Travis build

a-type committed:
Address concerns

- eliminate unecessary collection pooling
- restore validity of memory destroy tests
- test for cursor close in mongo
- test that disposed mongomapper cannot be used

a-type committed:
Expose MongoMapper on the Genesis library

a-type committed:
Export the MemoryMapper class on the library

jagoda committed:
Merge pull request #2 from a-type/fix-expose-mem

Export the MemoryMapper class on the library

Commit:
0.1.3

a-type committed:
Provide a more helpful error message when query model type is omitted

jagoda commented:
:-( rebase issues?

a-type commented:
I did rebase it. Odd.

jagoda committed:
Merge pull request #3 from a-type/helpful-find-error

Provide a more helpful error message when query model type is omitted

a-type committed:
Add helpful error messages to mongo find methods

a-type commented:
I messed up somewhere and rebased on master. I'm not sure how to fix it. I may open a PR against master and see if that works.
